File: play-2.5.x/K-Modules/kmodules/K-Base/app/K/Aop/annotations/Event.java
package K.Aop.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by kk on 15/8/25.
 */
@Target({ElementType.TYPE, ElementType.METHOD, ElementType.PARAMETER, ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
public @interface Event {
    String value() default "";
}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/K/Aop/annotations/Event.kt
package K.Aop.annotations

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Created by kk on 15/8/25.
 */
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.FIELD)
@Retention(RetentionPolicy.RUNTIME)
annotation class Event(val value: String = "")
